application/forms/EditNodeForm.php

```diff
@@ -457,4 +483,19 @@ public function isValid($data)
 
         return parent::isValid($data);
     }
+
+    public function hasDeleteButton()
+    {
+        return $this->deleteButtonName !== null;
+    }
+
+    public function shouldBeDeleted()
+    {
+        if (! $this->hasDeleteButton()) {
```

Comment:
If you have a static name for delete button, method `hasDeleteButton` is obsolate.

You can check if `getElement('delete_node')` is null instead.

Reply:
Since, the `deleteButtonName` property is required anyways. I will change the if condition to check if `deleteButtonName` property is set.

Reply:
Please set the `deleteButtonName` when declaring the property, it never changes after all.
protected $deleteButtonName = `delete_node`;